crates/cranelift/src/func_environ.rs

```diff
@@ -1627,11 +1646,13 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
         value: ir::Value,
         index: ir::Value,
     ) -> WasmResult<()> {
-        let pointer_type = self.pointer_type();
         let plan = &self.module.table_plans[table_index];
+        let table = plan.table;
+        let index_type = table.idx_type;
+        let index = self.cast_index_to_i64(&mut builder.cursor(), index, index_type);
```

Comment:
Like `table_get` above this'll want to avoid casting to i64 since some table sets are handled inline. This may need to propagate the `index_type` into various calls below so they can handle the 32/64 split.